### DISCUSSION TITLE
RQL "limit" doesn't work with "Show stored index fields only" in Studio

### DISCUSSION THREAD

**Discussion by krivulcik** [ravendb/ravendb#13265] [Q&A]
When using `limit` in RQL, turning on the "Show stored index fields only" option in Studio causes the following error:

Incorrect query command syntax: 4:3 Expected end of query but got: select

The query constructed is the following:

from index 'Product/Search'
limit 1
select __all_stored_fields

![image](https://user-images.githubusercontent.com/7997055/145828999-a98287d4-0ee8-466c-9d9c-43ae24b9faaa.png)

**Comment by ayende** [comment]
Yes, created at issue here: https://issues.hibernatingrhinos.com/issue/RavenDB-17695

We need to add the `select` before the limit.
